src/core/server/docs/kib_core_reviewing_so_pr.mdx

```diff
@@ -0,0 +1,67 @@
+---
+id: kibCoreReviewingSoPr
+slug: /kibana-dev-docs/review/reviewing-so-pr
+title: Reviewing SavedObject PRs
+description: How to review PRs that changes savedObjects registration
+date: 2022-09-30
+tags: ['kibana','dev', 'contributor', 'api docs']
+---
+
+# Reviewing SavedObject PRs
+
+## How do SO type automatic review assignment works?
+
+PRs modifying / adding / deleting any SO type registration will be flagged by the integration
+test located at `src/core/server/integration_tests/saved_objects/migrations/check_registered_types.test.ts`.
+
+This test will fail any time a potential migration-impactful change is performed on any SO type registration,
+and will force the PR's author to update the test's snapshot, which will trigger a review from the Core team.
+
+## What and how to review?
+
+Reviews will likely be triggered when one, or more, of these scenarios occur:
+- a new type was added (registered)
+- a type was removed (is no longer registered)
+- a new migration function was added
+- a mapping change was performed
+- a new validation schema was added
+
+### A new type was added
+
+We have another integration test detecting this scenario (`src/core/server/integration_tests/saved_objects/migrations/type_registrations.test.ts`)
+
+In that scenario, we should:
+- check the initial mappings of the type to spot potential issues:
+  - fields being defined explicitly in the mappings but not directly used for search
+  - overall amount of fields
+- check if the type is registered as `hidden: true` and encourage to do so otherwise. 
```

Comment:
```suggestion
- check if the type is registered as `hidden: true` and encourage to do so otherwise. 
  - this avoids polluting the global SO HTTP APIs with another type, and instead requires plugin developers to build their own HTTP APIs to access this type of SO if they truly need to
```